Bug 4247 syslog level

rohitkeshri1986 opened the pull request:
Ticket: #4247.
Added "Syslog level" checkbox in GUI.
User menu selection will be used as a filter for logs that are sent to remote server with "Syslog server" defined IP.
This does not affect local logging.

williambr commented:
Why is your migration numbered 103_*? It doesn't really make sense. In master the latest migration is 100 the next should be 101. What are the names of your 101 and 102 migrations?

rohitkeshri1986 commented:
0100_auto__chg_field_tunable_tun_value.py
0101_auto__del_registration__add_field_certificate_cert_chain__add_field_ce.py
0102_auto__del_field_settings_stg_sysloglevel.py
0103_auto__add_field_settings_stg_sysloglevel.py
To have more clarity i played with the GUI, so i have got some extra migration files.

Should i reinstall freeNAS from fresh, modify GUI  and and mail you the new migration william.

williambr commented:
Yes, thats preferable since your install is probably tainted.

williambr commented:
Why does your migration is changing system_certificateauthority, system_certificate and system_registration.

That doesn't make sense.
You should run the create migration with an update system with just your changes.

Also, please don't commit as "root".

rohitkeshri1986 commented:
I tried to create migration with an update system with just my changes by creating a user account.
As followes:

rohitkeshri1986 commented:
I tried to create migration with an update system with just my changes by creating a user account.

rohitkeshri1986 commented:
I tried to create migration with an update system with just my changes by creating a user account.
And ran following commands to create migration file.

rohit.keshri@freenas:/usr/local/www/freenasUI % sudo mount -uw /
rohit.keshri@freenas:/usr/local/www/freenasUI % sudo python manage.py schemamigration system --auto
 - Deleted model system.Registration
 + Added field cert_chain on system.Certificate
 + Added field cert_chain on system.CertificateAuthority
 + Added field stg_sysloglevel on system.Settings
Created 0101_auto__del_registration__add_field_certificate_cert_chain__add_field_ce.py. You can now apply this migration with: ./manage.py migrate system
rohit.keshri@freenas:/usr/local/www/freenasUI %

But i don't know why system.Registration is deleted and system.Certificate and system.CertificateAuthority gets modified 
as i have not done any modification in these fields.
Please suggest if i am doing anything wrong here, as i am new to this.

williambr commented:
No, thats the way to do it, but you should make a separate migration for the other ones that were forgotten.
I'll do that for you and let you know.

williambr commented:
There, wait for next nightly or make yourself a new build and now it should only create a migration with your changed fields.

rohitkeshri1986 commented:
Hi William, i have applied the changes on latest nightly and now i have got a migration that has only my modifications.

williambr commented on gui/system/models.py:
```diff
@@ -132,6 +132,17 @@ class Settings(Model):
+                    "server performs on behalf of the client. "
+                    "f_is_debug specify higher levels of debugging output. "
+                    "The default is f_info."),
+    )
```
Just to be sure here. Ar eyou sure the default syslog level (when no level is specified at all) is f_info?
Do you have any link to back it up? Because I would think the default is f_notice but I am not sure.

rohitkeshri1986 commented:
As per my understanding "f_info" suitable because earlier, syslog deamon used to flush all the logs to remote server. 
After modification syslog-ng should behave in similar fashion in default case. 

I could not find any universal definition or standard for default log level. This is decided on case basis. 
For example in Linux kernel, printk when a log level is not specified, the default log level is KERN_WARNING.
So i think we can too follow the stander that Linux follows but its optional.

williambr commented:
Ok, thats fine then. Thanks for clarifying.

rohitkeshri1986 commented:
Thank you William.
